discovery/gossiper.go

```diff
@@ -1872,17 +1876,24 @@ func (d *AuthenticatedGossiper) processRejectedEdge(
 		msg:    chanAnn,
 	})
 	if e1Ann != nil {
-		announcements = append(announcements, networkMsg{
-			source: d.selfKey,
-			msg:    e1Ann,
-		})
+		// Only add the ChanUpdate to the gossiper if the
+		// __dont_forward__ bit signals it.
+		if !e1Ann.MessageFlags.HasDontForward() {
```

Comment:
Let me see if I can make sense of this. This happens in the scenario where we are processing a rejected edge. The only time we add an edge to our graph that we didn't validate the AuthProof for is if we are one of the parties on the channel. I'm speculating because I'm still new to this segment of the codebase but it seems that this only happens when we add the edge prior to `announcement_signatures` and then possibly process the remote's `channel_update` prior to a valid `channel_announcement` which contains the proof. So in this scenario it actually stands to reason that this bit should *always* be set here, right?

Reply:
Good questions!
So as it is coded currently, both lnd peers will `readd` the channel after 6 confirmations are reached with the _dont_forward_bit unset here: https://github.com/lightningnetwork/lnd/pull/7759/files#diff-bbccb27bb17be513ad84c77ad8e37a4d76168956f7978309aab29a0cbfdf91acR3487 (for Scid-channels it works a bit different but the channel is readded as well => hence a new ChanUpdate is sent (directly to the peer) so we anticipate the dont_forward bit not set in the above case. However we can not be sure whether the peer ratelimited us or something and the ChanUpdate did not make it into its graph. So thats why I integrated this check to make sure we follow the network rules and don't gossip a ChanUpdate with the bit set, potentially getting punished by other peers for not sticking to the rules?